plugins/woocommerce/includes/rest-api/Controllers/Version3/class-wc-rest-products-controller.php

```diff
@@ -574,13 +583,19 @@ protected function set_product_images( $product, $images ) {
 					}
 
 					$attachment_id = wc_rest_set_uploaded_image_as_attachment( $upload, $product->get_id() );
+					$is_new_upload = true;
```

Comment:
Why don't we set the processed attachment ID here directly? At this point, the `$attachment_id` is always greater than `0`.

Reply:
Right, my bad, `wp_insert_attachment` can still return `0`. False alarm! 🙈